File: Criollo/Apps/CriolloApp/HelloWorldViewController.m
//
//  HelloWorldViewController.m
//  HelloWorld-ObjC
//
//  Created by Cătălin Stan on 11/23/15.
//  Copyright © 2015 Catalin Stan. All rights reserved.
//

#import "HelloWorldViewController.h"

@implementation HelloWorldViewController

- (NSString *)presentViewControllerWithRequest:(CRRequest *)request response:(CRResponse *)response {
    self.vars[@"title"] = NSStringFromClass(self.class);

    NSMutableString* text = [NSMutableString string];
    [text appendString:@"<h3>Request Enviroment:</h3><pre>"];
    [request.env enumerateKeysAndObjectsUsingBlock:^(NSString * _Nonnull key, NSString * _Nonnull obj, BOOL * _Nonnull stop) {
        [text appendFormat:@"%@: %@\n", key, obj];
    }];
    [text appendString:@"</pre>"];
    self.vars[@"text"] = text;

    return [super presentViewControllerWithRequest:request response:response];
}

@end

File: Criollo/Apps/CriolloTVApp/HelloWorldViewController.swift
//
//  HelloWorldViewController.swift
//  HelloWorld-Swift
//
//  Created by Cătălin Stan on 11/24/15.
//
//

import Criollo

class HelloWorldViewController: CRViewController {

    override func present(with request: CRRequest, response: CRResponse) -> String {
        self.vars["title"] = String(describing: type(of: self))

        var text:String = String()
        
        text += "<h3>Request Query:</h2><pre>"
        for (key, object) in request.query {
            text += "\(key): \(object)\n"
        }
        text += "</pre>"
        
        text += "<h3>Request Environment:</h2><pre>"
        for (key, object) in request.env {
            text += "\(key): \(object)\n"
        }
        text += "</pre>"
        
        self.vars["text"] = text
        return super.present(with: request, response: response)
    }

}
